posthog/api/plugin.py

```diff
@@ -0,0 +1,132 @@
+import json
+from typing import Any, Dict
+
+import requests
+from django.conf import settings
+from rest_framework import request, serializers, viewsets
+from rest_framework.decorators import action
+from rest_framework.exceptions import APIException
+from rest_framework.response import Response
+
+from posthog.models import Plugin, PluginConfig
+from posthog.plugins import download_plugin_github_zip, reload_plugins_on_workers
+
+
+class PluginSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Plugin
+        fields = ["id", "name", "description", "url", "config_schema", "tag", "from_json"]
+
+    def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Plugin:
+        if not settings.PLUGINS_INSTALL_FROM_WEB:
+            raise APIException("Plugin installation via the web is disabled!")
+        if len(Plugin.objects.filter(name=validated_data["name"])) > 0:
+            raise APIException('Plugin with name "{}" already installed!'.format(validated_data["name"]))
+        validated_data["archive"] = download_plugin_github_zip(validated_data["url"], validated_data["tag"])
+        if "from_json" in validated_data:  # prevent hackery
```

Comment:
Maybe just set `from_json` as `read_only` in the serializer? Cleaner that way

Reply:
done!